src/freqresp.jl

```diff
@@ -100,50 +123,87 @@ end
 Compute the magnitude and phase parts of the frequency response of system `sys`
 at frequencies `w`
 
-`mag` and `phase` has size `(length(w), ny, nu)`""" 
+`mag` and `phase` has size `(ny, nu, length(w))`""" 
 @autovec (1, 2) function bode(sys::LTISystem, w::AbstractVector)
     resp = freqresp(sys, w)
     return abs.(resp), rad2deg.(unwrap!(angle.(resp),1)), w
 end
-@autovec (1, 2) bode(sys::LTISystem) = bode(sys, _default_freq_vector(sys, Val{:bode}()))
+@autovec (1, 2) bode(sys::LTISystem; kwargs...) = bode(sys, _default_freq_lims(sys, Val{:bode}()); kwargs...)
+
+@autovec (1, 2) function bode(sys::LTISystem, lims::Tuple; kwargs...)
+    f = (w) -> begin
+        fr = eval_frequency(sys, w)
+        (abs.(fr), angle.(fr))
+    end
+    ys, grid = auto_grid(f, lims, (log10, exp10), (log10, identity); kwargs...)
+    angles = cat(ys[2]...,dims=3)
+    unwrap!(angles,3)
+    angles .= rad2deg.(angles)
+    cat(ys[1]...,dims=3), angles, grid
+end
 
 """`re, im, w = nyquist(sys[, w])`
 
 Compute the real and imaginary parts of the frequency response of system `sys`
 at frequencies `w`
 
-`re` and `im` has size `(length(w), ny, nu)`""" 
+`re` and `im` has size `(ny, nu, length(w))`""" 
 @autovec (1, 2) function nyquist(sys::LTISystem, w::AbstractVector)
     resp = freqresp(sys, w)
     return real(resp), imag(resp), w
 end
-@autovec (1, 2) nyquist(sys::LTISystem) = nyquist(sys, _default_freq_vector(sys, Val{:nyquist}()))
+@autovec (1, 2) function nyquist(sys::LTISystem, lims::Tuple; kwargs...)
+    # TODO check if better to only consider fr
+    f = (w) -> begin
+        fr = eval_frequency(sys, w)
+        (fr, real.(fr), imag.(fr))
+    end
+    ys, grid = auto_grid(f, lims, (log10, exp10), (identity,identity,identity); kwargs...)
+    return cat(ys[2]...,dims=3), cat(ys[3]...,dims=3), grid
+end
+@autovec (1, 2) nyquist(sys::LTISystem; kwargs...) = nyquist(sys, _default_freq_lims(sys, Val{:nyquist}()); kwargs...)
 
 """`sv, w = sigma(sys[, w])`
 
 Compute the singular values `sv` of the frequency response of system `sys` at
 frequencies `w`
 
-`sv` has size `(length(w), max(ny, nu))`""" 
+`sv` has size `(max(ny, nu), length(w))`""" 
 @autovec (1) function sigma(sys::LTISystem, w::AbstractVector)
     resp = freqresp(sys, w)
-    sv = dropdims(mapslices(svdvals, resp, dims=(2,3)),dims=3)
+    sv = dropdims(mapslices(svdvals, resp, dims=(1,2)),dims=2)
     return sv, w
 end
-@autovec (1) sigma(sys::LTISystem) = sigma(sys, _default_freq_vector(sys, Val{:sigma}()))
+# TODO: Not tested, probably broadcast problem on svdvals in auto_grid
+@autovec (1) function sigma(sys::LTISystem, lims::Tuple; kwargs...)
+    f = (w) -> begin
```

Comment:
```suggestion
    f = function (w)
```
This is a much nicer syntax to create anonymous function. Same goes in many places.